cmd/collection/main.go

```diff
@@ -581,6 +581,12 @@ func main() {
 			// register the manager for protocol events
 			node.ProtocolEvents.AddConsumer(manager)
 
+			// for collection nodes RPC inspectors must validate cluster prefixed topics
+			// using the ClusterIDSProvider methods implemented by the epoch manager.
+			for _, rpcInspector := range nodeBuilder.GossipSubConfig.RPCInspectors {
+				rpcInspector.SetClusterIDSProvider(manager)
```

Comment:
It appears that the PR needs to consider this comment ☝️

Reply:
If a non-collection node receives a cluster channel control message it will eventually be dropped and the sender will be punished. We also introduce a cluster prefix discard threshold, where we will process a certain number of cluster-prefixed control messages before we start to discard them. So at most a malicious node will be able to send up to the configured discard threshold amount of messages before that node is punished and pushed to the edge of the network. This allows nodes that have fallen behind in the protocol to have their messages processed.

Here we replace the polling pattern used by the ClusterIDSProvider in favor of the distributor-consumer pattern. ClusterID update events are distributed when new epoch components start and old epoch components stop. https://github.com/onflow/flow-go/pull/4189/commits/66f3aaca839fffe78f9992bdb598df333b293cdb

We add validation rpc inspectors as consumers to the ClusterIDUpdateDistributor https://github.com/onflow/flow-go/pull/4189/commits/7e988b8dec9e03d07f62c2df78371acefaa095f6